tests/unit/semantic_release/hvcs/test_github.py

```diff
@@ -539,16 +531,68 @@ def test_create_or_update_release_when_create_fails_and_no_release_for_tag(
         mock_edit_release_notes.assert_not_called()
 
 
+def test_asset_upload_url(default_gh_client):
+    release_id = 1
+    # '{?name,label}' are added by github.com at least, maybe custom too
+    # https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release
+    resp_payload = {
+        "upload_url": (
+            f"{default_gh_client.api_url.replace('api', 'uploads')}/repos/"
+            f"{default_gh_client.owner}/{default_gh_client.repo_name}/"
+            f"releases/{release_id}/"
+            "assets{?name,label}"
+        ),
+        "status": "success",
+    }
+    with requests_mock.Mocker(session=default_gh_client.session) as m:
+        m.register_uri("GET", github_api_matcher, json=resp_payload, status_code=200)
+        assert default_gh_client.asset_upload_url(
+            release_id
+        ) == "https://uploads.{domain}/repos/{owner}/{repo}/releases/{release_id}/assets".format(
```

Comment:
Here could also be simpler if you could do `f"https://{domain}...".format(domain=Github.ASSET_UPLOAD_DOMAIN, ...)`